modules/wordads/php/class-wordads-california-privacy.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+use Automattic\Jetpack\Assets;
+
+/**
+ * Class WordAds_California_Privacy
+ *
+ * Implementation of [California Consumer Privacy Act] (https://leginfo.legislature.ca.gov/faces/codes_displayText.xhtml?lawCode=CIV&division=3.&title=1.81.5.&part=4.&chapter=&article=) as applicable to WordAds.
+ * Includes:
+ * - Do Not Sell My Personal Information shortcode and widget.
+ * - Modal notice to toggle opt-in/opt-out.
+ * - Cookie handling. Implements IAB usprivacy cookie specifications.
+ * - Client side geo-detection of California visitors by IP address. Avoids issues with page caching.
+ */
+class WordAds_California_Privacy {
+
+	/**
+	 * Initializes required scripts and shortcode.
+	 */
+	public static function init() {
+
+		// Initialize required scripts.
+		add_action( 'wp_enqueue_scripts', array( __CLASS__, 'enqueue_scripts' ) );
+
+		// Initialize shortcode.
+		self::init_shortcode();
+	}
+
+	/**
+	 * Enqueue required CCPA JavaScript on the frontend.
+	 */
+	public static function enqueue_scripts() {
+		wp_enqueue_script(
+			'wordads_ccpa',
+			Assets::get_file_url_for_environment(
+				'_inc/build/wordads/js/wordads-ccpa.min.js',
+				'modules/wordads/js/wordads-ccpa.js'
+			),
+			array(),
+			'1.0.0',
+			true
+		);
+
+		wp_localize_script(
+			'wordads_ccpa',
+			'ccpaSettings',
+			array(
+				'defaultOptinCookieString' => esc_html( self::get_optin_cookie_string() ),
+				'cleanslateUrl'            => esc_url( WORDADS_URL ) . 'css/cleanslate.css',
+				'ccpaCssUrl'               => esc_url( WORDADS_URL ) . 'css/wordads-ccpa.min.css',
```

Comment:
I wasn't able to find an example of this for SCSS/CSS files, only Javascript. If you know how this could be easily done I'll add it.

Reply:
It works the same way as for JS files, something like this:
https://github.com/Automattic/jetpack/blob/78c3a2bb05be76c0aea8eaf419ee45d85fefa969/class-jetpack-wizard-banner.php#L114-L117

Reply:
I was able to get this working, but it leaves build generated `*.css` files in the `/modules/wordads/css` folder that now need to be added to `.gitignore`. I don't see other modules working this way. Is there a more "Jetpack" way of having module specific SCSS?

Reply:
> it leaves build generated *.css files in the /modules/wordads/css folder that now need to be added to .gitignore. 

I'm not sure I understand. Does the build generate .css files that we do not want to get shipped with Jetpack's stable version?

Reply:
>I'm not sure I understand. Does the build generate .css files that we do not want to get shipped with Jetpack's stable version?

Having the generated .css shipped isn't a problem, but I didn't necessarily want it committed to Git, which would require a `.gitignore` change. The `/modules/wordads/css` folder would then have a mixture of SCSS,  handcrafted css files I do want in the repo (`styles.css`), and autogenerated ones that I don't (`wordads-ccpa.css`, `wordads-ccpa.min.css`).

Maybe it makes sense to output the generated CSS files to the `/css` folder which is already part of the `.gitignore` instead of outputting them to the `/modules/wordads/css` folder?

Reply:
> Maybe it makes sense to output the generated CSS files to the /css folder which is already part of the .gitignore instead of outputting them to the /modules/wordads/css folder?

Yes, that's a good point and a good idea. This may be a better approach. Moving the SCSS to the `/scss` folder and thus the CSS to the `/css` folder should simplify things in terms of things to maintain and run.